src/Core/References/References.cs

```diff
@@ -184,10 +184,16 @@ public async Task AddPackage(string name, Action<string>? statusCallback = null)
 
         private void Reset()
         {
+            var oldMetadata = _metadata;
```

Comment:
Is this a fix to some of the build issues?

Reply:
Yes this is a fix that @cgranade added to fix the tests running out of memory during builds.